### ISSUE TITLE
Conditional in included files from ./handlers/main.yml work only for the first include

### ISSUE THREAD

**Issue by guits** [ansible/ansible#23389]
Hi,

I'm having a strange behaviour with conditionnal on `block` statement.

I'm trying to trigger multiple handlers (using `listen` statement) from a task that notifies several listeners.
The handlers are in separate files that are included from `./roles/test/handlers/main.yml` 
I'm using `block` statement with conditionnal for each group of handler, it seems only the first `block` has its `when` evaluated.

#### ISSUE TYPE
Bug report

#### COMPONENT NAME
It appears on `block` feature used in `handlers`

#### ANSIBLE VERSION
    ansible-playbook 2.2.1.0
      config file =
      configured module search path = Default w/o overrides

#### CONFIGURATION
No custom configuration

#### OS / ENV

CentOS 7, x86_64, Python 2.7.13

#### INVENTORY HOSTS FILE

    [group1]
    laptop ansible_host=127.0.0.1

    [group2]
    
    [group3]

##### FIRST TEST

./roles/test/handlers/main.yml : 
```yaml
---
- include: ./group1.yml

- include: ./group2.yml

- include: ./group3.yml
```

./roles/test/handlers/group1.yml :
```yaml
---
- block:
  - name: first block
    debug:
      msg: "first block, first task"
    listen: "first block"
  - name: first block
    debug:
      msg: "first block, second task"
    listen: "first block"
  when:
    - "'group1' in group_names"
```

./roles/test/handlers/group2.yml :
```yaml
---
- block:
  - name: second block
    debug:
      msg: "second block, first task"
    listen: "second block"
  - name: second block
    debug:
      msg: "second block, second task"
    listen: "second block"
  when:
    - "'group2' in group_names"
```

./roles/test/handlers/group3.yml :
```yaml
---
- block:
  - name: third block
    debug:
      msg: "third block, first task"
    listen: "third block"
  - name: third block
    debug:
      msg: "third block, second task"
    listen: "third block"
  when:
    - "'group3' not in group_names"
```

./roles/test/tasks/main.yml :
```yaml
---
- name: dummy task
  debug:
    msg: "dummy task"
  changed_when: true
  notify:
    - "first block"
    - "second block"
    - "third block"
```

#### EXPECTED RESULT
```bash
TASK [test : dummy task] *******************************************************
ok: [laptop] => {
    "msg": "dummy task"
}

RUNNING HANDLER [test : first block] *******************************************
ok: [laptop] => {
    "msg": "first block, first task"
}

RUNNING HANDLER [test : first block] *******************************************
ok: [laptop] => {
    "msg": "first block, second task"
}

RUNNING HANDLER [test : second block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : second block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : third block] *******************************************
ok: [laptop] => {
    "msg": "third block, first task"
}

RUNNING HANDLER [test : third block] *******************************************
ok: [laptop] => {
    "msg": "third block, second task"
}
```

#### ACTUAL RESULT
```bash
TASK [test : dummy task] *******************************************************
ok: [laptop] => {
    "msg": "dummy task"
}

RUNNING HANDLER [test : first block] *******************************************
ok: [laptop] => {
    "msg": "first block, first task"
}

RUNNING HANDLER [test : first block] *******************************************
ok: [laptop] => {
    "msg": "first block, second task"
}

RUNNING HANDLER [test : second block] ******************************************
ok: [laptop] => {
    "msg": "second block, first task"
}

RUNNING HANDLER [test : second block] ******************************************
ok: [laptop] => {
    "msg": "second block, second task"
}

RUNNING HANDLER [test : third block] ******************************************
ok: [laptop] => {
    "msg": "third block, first task"
}

RUNNING HANDLER [test : third block] ******************************************
ok: [laptop] => {
    "msg": "third block, second task"
}
```

#### EXPECTED RESULT
```bash
TASK [test : dummy task] *******************************************************
ok: [laptop] => {
    "msg": "dummy task"
}

RUNNING HANDLER [test : first block] *******************************************
ok: [laptop] => {
    "msg": "first block, first task"
}

RUNNING HANDLER [test : first block] *******************************************
ok: [laptop] => {
    "msg": "first block, second task"
}

RUNNING HANDLER [test : second block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : second block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : third block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : third block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}
```

#### ACTUAL RESULT
```bash
TASK [test : dummy task] *******************************************************
ok: [laptop] => {
    "msg": "dummy task"
}

RUNNING HANDLER [test : first block] ******************************************
ok: [laptop] => {
    "msg": "first block, first task"
}

RUNNING HANDLER [test : first block] ******************************************
ok: [laptop] => {
    "msg": "first block, second task"
}

RUNNING HANDLER [test : second block] ******************************************
ok: [laptop] => {
    "msg": "second block, first task"
}

RUNNING HANDLER [test : second block] ******************************************
ok: [laptop] => {
    "msg": "second block, second task"
}

RUNNING HANDLER [test : third block] ******************************************
ok: [laptop] => {
    "msg": "third block, first task"
}

RUNNING HANDLER [test : third block] ******************************************
ok: [laptop] => {
    "msg": "third block, second task"
}
```

#### EXPECTED RESULT
```bash
TASK [test : dummy task] *******************************************************
ok: [laptop] => {
    "msg": "dummy task"
}

RUNNING HANDLER [test : first block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : first block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : second block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : second block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : third block] ******************************************
ok: [laptop] => {
    "msg": "third block, first task"
}

RUNNING HANDLER [test : third block] ******************************************
ok: [laptop] => {
    "msg": "third block, second task"
}
```

#### ACTUAL RESULT
```bash
TASK [test : dummy task] *******************************************************
ok: [laptop] => {
    "msg": "dummy task"
}

RUNNING HANDLER [test : first block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : first block] *******************************************
skipping: [laptop] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

RUNNING HANDLER [test : second block] ******************************************
ok: [laptop] => {
    "msg": "second block, first task"
}

RUNNING HANDLER [test : second block] ******************************************
ok: [laptop] => {
    "msg": "second block, second task"
}

RUNNING HANDLER [test : third block] ******************************************
ok: [laptop] => {
    "msg": "third block, first task"
}

RUNNING HANDLER [test : third block] ******************************************
ok: [laptop] => {
    "msg": "third block, second task"
}
```

It seems only the first include is working.
I was more expecting that either all include should works or nothing, not just the first one.

**Comment by fridim** [comment]
I successfully reproduced the issue with this simplified playbook:
https://gist.github.com/fridim/afdbd71d021c48d14bdf899b1dd53c89

only first block is skipped.

tested with :
```
ansible 2.2.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

and  devel
```
ansible 2.4.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/fridim/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible-2.4.0-py3.6.egg/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.0 (default, Jan 16 2017, 12:12:55) [GCC 6.3.1 20170109]
```

FWIW, this playbook without includes works for both versions:
https://gist.github.com/fridim/bf3cb4ca16cfeb5f6ed0ff270355efa6

Also tested  with includes and without blocks, no problem.

**Comment by guits** [comment]
@bcoca any update on this?

**Comment by bcoca** [comment]
cc @jimi-c

**Comment by guits** [comment]
@jimi-c ?

**Comment by mkrizek** [comment]
Fixed in #35921.
